src/drivers/camera_trigger/camera_trigger.cpp

```diff
@@ -477,6 +482,23 @@ CameraTrigger::cycle_trampoline(void *arg)
 				cmd_result = vehicle_command_s::VEHICLE_CMD_RESULT_ACCEPTED;
 
 			}
+
+		} else if (cmd.command == 12345 /* VEHICLE_CMD_SEND_CAMERA_IMAGE_META */) {
+
+			// This command is designed to allow the GCS / ground to re-request
+			// trigger packets that might have dropped over the link.
+			// The GCS can infer this when the trigger sequence skips once. It should
+			// store the triggers on the ground in a hash map using the sequence as
+			// hash. Then simply checking the last received index allows for an
+			// efficient retransmission logic.
+
+			if (static_cast<int>(cmd.param1) == 0 /*_prev_image.seq*/) {
+				// Send prev image again
```

Comment:
The robustness judgement should be made on timing, not on queue size. You want to allow for ~50-250 ms retransmission requests for the GCS before it gets overwritten. Taking into account that we realistically will not exceed 3 Hz with current systems that means we 250 / 333 ms = 0.75 messages queue size, rounding up to 1.